lib/snapshot/utils.go

```diff
@@ -178,6 +178,37 @@ func CreateTarFromDirectory(target, dir string) error {
 	})
 }
 
+// CreateDirectoryFromTar creates a directory containing the contents of the given
+// tar archive.
+func CreateDirectoryFromTar(target, tar string) error {
```

Comment:
this function should be in tario package.
Initiate `NewGzipReader`, use `.Next()` to iterate through items like in [UpdateFromTarReader](https://github.com/uber/makisu/blob/c75663b197c972ff65999db79c36eae6f622b91d/lib/snapshot/mem_fs.go#L156). No need to care too much about actual content.